chart/templates/configmap-etcd-bootstrap.yaml

```diff
@@ -0,0 +1,39 @@
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: etcd-bootstrap-{{ .Values.role }}
+  namespace: {{ .Release.Namespace }}
+  labels:
+    app: etcd
+data:
+  bootstrap.sh: |-
+    #!/bin/sh
+    while true;
+    do
+      wget http://localhost:8080/initialization/status -S -O status;
+      STATUS=`cat status`;
+      case $STATUS in
+      "New")
+            wget http://localhost:8080/initialization/start -S -O - ;;
+      "Progress")
+            sleep 1;
+            continue;;
+      "Failed")
+            continue;;
+      "Successful")
+            exec etcd --data-dir=/var/etcd/data \
```

Comment:
I was anyways thinking of using `--config-file` parameter pointing to [etcd.conf.yaml](https://github.com/etcd-io/etcd/blob/master/etcd.conf.yml.sample) which looks better. So lets add another confimap data field called `etcd.conf.yaml` and move the parameters there. We will probably mount this file under `/var/etcd/config`. Also, `bootstrap.sh` file under `/var/etcd/bin`

Reply:
I agree on using etcd.conf.yaml is a better approach and I was actually about to compleet this change request. But I just realised that this file's structure seems to be in sync with current [examples](https://github.com/gardener/etcd-backup-restore/tree/master/example). Once we have the chart in the repo, we'll need to update also the chart with each change in the examples.

Doing the suggested changes will diverge the structure in the examples, and will make the chart relatively harder to maintain in long term. Rather than having chart best practices, at this stage it seems to be more important to me to keep the chart maintenance more hassle-free in short term. We can start doing such improvements after seeing the chart does worth spending such maintenance overhead. I'm just concerned about keeping the chart up-to-date as the project evolves.

Here are the recent changes happened in the examples directory, there seems to be monthly update on examples for a while, i'm not sure if that will change soon:
```
$ git lg example/
* 6712697 - Update examples (4 months ago) <Swapnil Mhamane>
* c45e7b3 - Update changelog (5 months ago) <Swapnil Mhamane>
* ae89094 - Add generate-example script (5 months ago) <Swapnil Mhamane>
* 7711680 - Fix to make sure etcd is not running when the backup container is failing to take a backup to cloud. Doesnt need to restart the backup container but keep retrying instead keeping etcd down. (5 months ago) <I308301>
* 32a226b - TLS Support for snapshotter. (6 months ago) <I308301>
* 4d03f04 - Fix: check member directory instead of data directory. Cannot delete data directory as its the mount point. (6 months ago) <I308301>
* 779f6b3 - Changed data directory deletion to delete contents in the directory instead of the directory. (6 months ago) <I308301>
* 926f0ad - Remove dependency of gcs snapstore on project-id enviornment variable (7 months ago) <Swapnil Mhamane>
* ddd683c - GCS object store support. (7 months ago) <I308301>
* a1301a5 - Add store prefix option (7 months ago) <Swapnil Mhamane>
* b7e95fc - Asynchronize http call for initialization (7 months ago) <Swapnil Mhamane>
* 0e0ad3a - Expose etcd initialiation functionaliy over http (7 months ago) <Swapnil Mhamane>
* c4449af - Adding comments in example statefulset yaml. (8 months ago) <I308301>
* 7bb28ba - Integrating restore code. (8 months ago) <I308301>
* 52ed7bd - Refactor code to have single root command (8 months ago) <Swapnil Mhamane>
* e8bb35b - Add restore support (8 months ago) <Swapnil Mhamane>
```

Would you still prefer me to proceed with this change?

Reply:
Oh, i see the examples are generated from [hack/templates/etcd-statefulset.yaml.tpl](https://github.com/gardener/etcd-backup-restore/blob/master/hack/templates/etcd-statefulset.yaml.tpl), I think it would be even better to make this file a synchronized structural copy of it.

Reply:
Once we moved to chart, I don't feel need of maintaining separate example yamls. You can simply remove those. So, in my opinion you can go ahead and use `etcd.conf.yaml`.

Reply:
Cool, thanks for the clarification, i'll follow up.